tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -788,9 +790,15 @@ SpirvDebugLexicalBlock *
 SpirvBuilder::createDebugLexicalBlock(SpirvDebugSource *source, uint32_t line,
                                       uint32_t column,
                                       SpirvDebugInstruction *parent) {
+  assert(insertPoint && "null insert point");
   auto *inst =
       new (context) SpirvDebugLexicalBlock(source, line, column, parent);
   module->addDebugInfo(inst);
+  if (insertPoint->empty()) {
+    insertPoint->setDebugScope(new (context) SpirvDebugScope(inst));
+  } else {
+    insertPoint->addInstruction(new (context) SpirvDebugScope(inst));
```

Comment:
Why is DebugScope added as an explicit instruction in cases where the current basic block is empty?
I think this should not be needed because `SpirvBasicBlock::invokeVisitor` is manually visiting `debugScope` member variable. So I think in both cases (whether BB is empty or not), calling `setDebugScope` should be enough. Am I missing something?

Reply:
Oh I see. Yeah that makes sense. Thanks for explanation.